src/frontend/src/tests/lib/utils/token.utils.spec.ts

```diff
@@ -275,3 +283,240 @@ describe('sumUsdBalances', () => {
 		expect(sumUsdBalances([undefined, undefined])).toBeUndefined();
 	});
 });
+
+describe('mapDefaultTokenToToggleable', () => {
+	describe('Every other random token is only enabled if the user enables it', () => {
+		it('should enable LINK if user enables it', () => {
+			const result = mapDefaultTokenToToggleable({
+				defaultToken: LINK_TOKEN,
+				userToken: { ...LINK_TOKEN, enabled: true }
+			});
+
+			expect(result.enabled).toEqual(true);
+		});
+
+		it('should not enable LINK if user has not enabled it', () => {
+			const result = mapDefaultTokenToToggleable({
+				defaultToken: LINK_TOKEN,
+				userToken: { ...LINK_TOKEN, enabled: false }
+			});
+
+			expect(result.enabled).toEqual(false);
+		});
+
+		it('should not enable LINK if userToken is undefined', () => {
+			const result = mapDefaultTokenToToggleable({
+				defaultToken: LINK_TOKEN,
+				userToken: undefined
+			});
+
+			expect(result.enabled).toEqual(false);
+		});
+	});
+
+	describe('Default ICRC tokens are always enabled', () => {
+		describe('ckBTC', () => {
+			const dummyCkBTC = { ledgerCanisterId: IC_CKBTC_LEDGER_CANISTER_ID } as IcCkToken;
```

Comment:
i think there is a `mockValidIcToken` mock variable

you can expand that to avoid casting